firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -103,4 +124,105 @@ String getApplicationId() {
   String getName() {
     return firebaseApp.getName();
   }
+
+  /**
+   * Create firebase installation id for the {@link FirebaseApp} on FIS Servers and client side
+   * cache.
+   *
+   * <pre>
+   *     The workflow is:
+   *         check if cache empty or cache status is REGISTER_ERROR
+   *                                 |
+   *                           create random fid
+   *                                 |
+   *               update cache with cache status UNREGISTERED
```

Comment:
is there a status "IN PROGRESS" or something to prevent multiple requests in parallel?
do you get a lock on the database entry or how do you deal with multithreading?

Reply:
We currently don't have any IN PROGRESS state. I am looking at ios-sdk to see how its handled. I ll discuss with you in person about this.